packages/server/lib/browsers/firefox.ts

```diff
@@ -422,6 +451,15 @@ export async function open (browser: Browser, url: string, options: BrowserLaunc
     ],
   })
 
+  // CDP is deprecated in Firefox 129 and up.
+  // In order to enable CDP (without BiDi), we need to set
+  // remote.active-protocol=2
+  // To enable BiDi (without CDP), we need to set
+  // remote.active-protocol=1
+  // @see https://fxdx.dev/deprecating-cdp-support-in-firefox-embracing-the-future-with-webdriver-bidi/
+  // @see https://github.com/cypress-io/cypress/issues/29713
+  defaultLaunchOptions.preferences['remote.active-protocols'] = USE_WEBDRIVER_BIDI ? 1 : 2
```

Comment:
Could maybe make constants out of these?

```suggestion
  defaultLaunchOptions.preferences['remote.active-protocols'] = USE_WEBDRIVER_BIDI ? ENABLE_BIDI_WITHOUT_CDP : ENABLE_CDP_WITHOUT_BIDI
```

Reply:
thats a good idea. Makes the intent clear

Reply:
did something similar in 804f524a46. How does that look?